Modules/_cursesmodule.c

```diff
@@ -387,6 +399,104 @@ PyCurses_ConvertToString(PyCursesWindowObject *win, PyObject *obj,
     return 0;
 }
 
+static int
+color_converter(PyObject *arg, void *ptr)
+{
+    long color_number;
+    int overflow;
+
+    color_number = PyLong_AsLongAndOverflow(arg, &overflow);
+    if (PyErr_Occurred())
+        return 0;
+
+    if (color_number < 0) {
```

Comment:
It is -1 if the argument is larger than LONG_MAX.

I would write:
```c
if (overflow > 0 || color_number > COLORS) {
    ...
}
else if (overflow < 0 || color_number < 0) {
    ...
}
```
Please add also tests which cover the following cases: `-1`, `COLORS+1`, `-2**31-1`, `2**31`, `-2**63-1`, `2**63`.